tools/flaky_test_monitor/level1/process_summary1_results_test.go

```diff
@@ -101,58 +154,24 @@ func runProcessSummary1TestRun(t *testing.T, jsonExpectedActualFile string) {
-		actualPackageResults := actualTestRun.PackageResults[packageIndex]
-
-		require.Equal(t, expectedPackageResults.Elapsed, actualPackageResults.Elapsed)
-		require.Equal(t, expectedPackageResults.Package, actualPackageResults.Package)
-		require.Equal(t, expectedPackageResults.Result, actualPackageResults.Result)
-		require.Empty(t, expectedPackageResults.TestMap, actualPackageResults.TestMap)
-
-		// check outputs of each package result
-		require.Equal(t, len(expectedPackageResults.Output), len(actualPackageResults.Output))
-		for packageOutputIndex := range expectedPackageResults.Output {
-			require.Equal(t, expectedPackageResults.Output[packageOutputIndex], actualPackageResults.Output[packageOutputIndex])
-		}
-
-		// check tests results of each package
-		checkTestResults(t, expectedPackageResults.Tests, actualPackageResults.Tests)
-	}
-	// finally, compare the entire actual test run against what's expected - if there were any discrepancies they should have been caught by now
-	require.Equal(t, expectedTestRun, actualTestRun)
-}
-
-func checkTestResults(t *testing.T, expectedTestResults []common.TestResult, actualTestResults []common.TestResult) {
-	require.Equal(t, len(expectedTestResults), len(actualTestResults))
-	for testResultIndex := range expectedTestResults {
-
-		expectedTestResult := expectedTestResults[testResultIndex]
-		actualTestResult := actualTestResults[testResultIndex]
+func checkLevel1Summaries(t *testing.T, expectedLevel1Summary common.Level1Summary, actualLevel1Summary common.Level1Summary) {
+	// number of results should be the same between expected and actual
+	require.Equal(t, len(expectedLevel1Summary.Rows), len(actualLevel1Summary.Rows))
 
-		// check all outputs of each test result
-		require.Equal(t, len(expectedTestResult.Output), len(actualTestResult.Output), fmt.Sprintf("TestResult[%d].Test: %s", testResultIndex, actualTestResult.Test))
-		for testResultOutputIndex := range expectedTestResult.Output {
-			require.Equal(t, expectedTestResult.Output[testResultOutputIndex], actualTestResult.Output[testResultOutputIndex], fmt.Sprintf("TestResult[%d] Output[%d]", testResultIndex, testResultOutputIndex))
-		}
+	// check that all expected results are in actual results
+	for actualRowIndex := range actualLevel1Summary.Rows {
+		require.Contains(t, actualLevel1Summary.Rows, expectedLevel1Summary.Rows[actualRowIndex], printLevel1TestResult(expectedLevel1Summary.Rows[actualRowIndex].TestResult, "expected not in actual"))
+	}
 
-		require.Equal(t, expectedTestResult.Package, actualTestResult.Package)
-		require.Equal(t, expectedTestResult.Test, actualTestResult.Test)
-		require.Equal(t, expectedTestResult.Elapsed, actualTestResult.Elapsed, fmt.Sprintf("TestResult[%d].Test: %s", testResultIndex, actualTestResult.Test))
-		require.Equal(t, expectedTestResult.Result, actualTestResult.Result)
+	// check that all actual results are in expected results
+	for exptedRowIndex := range expectedLevel1Summary.Rows {
```

Comment:
similar comment here

Reply:
Yes, we want to test that all actual level 1 result rows are in the actual level 1 result rows and vice versa.